myFitnessTrainer/src/screens/WorkoutScreen.js

```diff
@@ -1,11 +1,99 @@
-import { StyleSheet, View } from "react-native";
+import { useState } from 'react';
+import { StyleSheet, View, Dimensions, Text } from "react-native";
+import Carousel from 'react-native-reanimated-carousel';
+import Dots from 'react-native-dots-pagination';
 
 import EndWorkoutButton from "../components/EndWorkoutButton";
+import { GestureHandlerRootView } from 'react-native-gesture-handler';
+import * as Progress from 'react-native-progress';
+
+const width = Dimensions.get('window').width;
+
+// FOR TESTING, WILL USE THE DAILY EXERCISES FROM DB
+const testExercise = {
+    default_exercise_stat: null,
+    description: "this is the description",
+    equipment: "test equipment",
+    fitness_goal: "strength",
+    fitness_level: "beginner",
+    muscle_group: "lower body",
+    name: "test exercise name",
+    video: {
+        female: "www.google.com",
+        male: "www.google.com"
+    }
+};
+// FOR TESTING, USE THE DAILY EXERCISES
+const testExercisesDay0 = [testExercise,testExercise,testExercise,testExercise];
 
 const WorkoutScreen = () => {
+    const workoutLength = testExercisesDay0.length;
+    const [exerciseIndex, setExerciseIndex] = useState(0);
+    const [progress, setProgress] = useState(1/workoutLength);
+    const height = Dimensions.get('window').height;
+
+    // Determines if exercise is complete
+    const displayViewWorkoutSummary = (index) => {
+        if (index === workoutLength - 1) {
+            return (
+                <EndWorkoutButton />
```

Comment:
I think this button should show up in a new screen after the last workout. With this we can keep all the cards standardized

In the new screen, I think there can be a "finish" button so that the user can officially finish the workout (just in case they happen to swipe forward accidentally). And after the "finish" button is pressed, the "view workout summary" button can appear / replace the finish button

Reply:
I like that!

Reply:
made one change to your idea - when they press the 'finish workout' button, it take them to the workout summary screen.